x/concentrated-liquidity/swaps.go

```diff
@@ -338,7 +338,7 @@ func (k Keeper) computeOutAmtGivenIn(
 		amountCalculated:         sdk.ZeroDec(),                  // tokenOut
 		sqrtPrice:                curSqrtPrice,
 		// Pad (or don't pad) current tick based on swap direction to avoid off-by-one errors
-		tick:                     swapStrategy.InitializeTickValue(p.GetCurrentTick()),
+		tick:                     p.GetCurrentTick(),
```

Comment:
When I change this to `tick:                     p.GetCurrentTick() + 1` to see the repro, nothing breaks.

Can we add a test that ensures that prevents this issue from being reintroduced?

Reply:
From the issue desc:
> I had a local repro hitting this specific case for the second bug at a higher level of abstraction with swaps but did not want to block this PR on manually calculating the test vectors. Will include this test in a future PR with functional tests.

This will also be covered thoroughly at the level of abstraction of swaps in #5430, which I don't think is worth blocking this PR on. If you add 1 to the current tick input in `InitializeNextTickIterator`, tests should break.